src/hotspot/share/runtime/os.cpp

```diff
@@ -955,9 +996,7 @@ void os::print_hex_dump(outputStream* st, address start, address end, int unitsi
   // Print out the addresses as if we were starting from logical_start.
   st->print(PTR_FORMAT ":   ", p2i(logical_p));
   while (p < end) {
-    if (is_readable_pointer(p)) {
-      print_hex_readable_pointer(st, p, unitsize);
-    } else {
+    if (!print_hex_location(st, p, unitsize)) {
```

Comment:
Pull the `??????` printout straight into `print_hex_location` and remove the need for the return value?